internal/util/diff/diff.go

```diff
@@ -320,22 +324,68 @@ func (d *defaultPkgDiffer) Diff(pkgs ...string) error {
 
 // prepareForDiff removes metadata such as .git and Kptfile from a staged package
 // to exclude them from diffing.
-func (d *defaultPkgDiffer) prepareForDiff(dir string) error {
+func (d *defaultPkgDiffer) prepareForDiff(dir types.FileSystemPath) error {
 	excludePaths := []string{".git", kptfilev1.KptFileName}
 	for _, path := range excludePaths {
-		path = filepath.Join(dir, path)
-		if err := os.RemoveAll(path); err != nil {
+		rmpath := filepath.Join(dir, path)
+		if err := os.RemoveAll(rmpath); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// PkgGetter knows how to fetch a package given a git repo, path and ref.
+type PkgGetter interface {
```

Comment:
@martinmaly This should address the concerns we were discussing. This is similar to Evaluator interface.